RepositoryCompilerTests/Unit/CodeModelTestDataFactory.cs

```diff
@@ -162,6 +162,138 @@ private Repository GetRepository()
             };
         }
 
+        public IEnumerable<string> GetATFDMultipleClassTexts()
+        {
+            return new[]
+            {
+                @"
+            using System.Collections.Generic;
+            namespace NDCApp.Model.Data
+            {
+                public class Class1
+                {
+                    private int a1;
+                    private int a2;
+                   
+                    private bool m1(){
+                        Class2 class2 = new Class2();
+                        if(class2.a1 == class2.a2 == a1){
+                            return true;
+                        }
+                        return false;
+                    }
+                   
+                }
+            };
+                public class Class2
+                {
+                    public int a1;
+                    public int a2;
+                    private int a3;
+                    private int a4;
+                };
+            }",
+                @"
+            using System.Collections.Generic;
+            namespace NDCApp.Model.Data
+            {
+                  public class Class3
+                  {
+                   
+                    public double m1(){
+                        Class4 class4 = new Class4();
+                       
+                        class4.Hours = 23;
```

Comment:
I like the test data suite for the ATFD, but I think it's missing an important case when we have multiple methods in a class, as well as when we are referencing multiple classes and their fields. Such a case is especially important if we change the ATFD calculation.

Maybe change classes 3 and 4 to reference some subset of the other 6 classes in some way, as I believe classes 5 and 6 cover the test case that you currently have with 3 and 4.